### ISSUE TITLE
Help Wanted - Control Module

### ISSUE THREAD

**Issue by MuhammadAlhaddad** [ApolloAuto/apollo#13330]
Hello!
I have asked before about how to launch control module with MPC controller? and I have searched for answer but I didn't find it.
Actually I have tried to launch MPC controller using: 
`cyber_launch start modules/control/launch/control_mpc_control.launch`
But I have found that it still run LON+LAT controllers.
Any answer is appreciated.

**Comment by daohu527** [comment]
Could you change 
```
active_controllers: LAT_CONTROLLER
active_controllers: LON_CONTROLLER
```
to
```
active_controllers: MPC_CONTROLLER
```
and have a try : )

**Comment by MuhammadUbuntu** [comment]
@daohu527 Thank for reply. I have changed it. The problem that I have found is:
1- There is no configuration for mpc in `control_conf.pb.txt` that I have (I don't know why) so I made (Copy/Paste) for the whole file from github and replaced` active_controllers` as you have mentioned. 
Then I run it again and when I made debug for control module I have found that it computes` Local_view` and doesn't enter controller_agent at all.
2- also I have tried `cyber_launch start modules/control/launch/control_mpc_control.launch ` and I have gotten the same result. 
next I show you debug process.
![Screenshot from 2021-01-19 15-30-54](https://user-images.githubusercontent.com/73900127/105036095-0ebaf280-5a6d-11eb-9fbf-eb8c44accd43.png)

**Comment by daohu527** [comment]
Pls check `use_control_submodules` is false.  

If you run with `cyber_launch start modules/control/launch/control.launch` it will init the mpc contro moudle. Considering that your configuration file is incomplete, I suggest you check the configuration and code first

**Comment by MuhammadUbuntu** [comment]
@daohu527 Thanks for reply. 
I have cloned apollo again and I have kept Lat+Lon controller in control_conf.pb.txt and I have checked that `use_control_submodules` is false. 
For debugging the control module (the goal is to see what is the used controller in control_agent) I have added two breakpoints in file control_component.cc at lines 161 and 244.
![Screenshot from 2021-01-21 14-25-12](https://user-images.githubusercontent.com/73900127/105344904-9ed77400-5bf4-11eb-8c3f-9132d1959054.png)
![Screenshot from 2021-01-21 14-22-48](https://user-images.githubusercontent.com/73900127/105344716-5c159c00-5bf4-11eb-88ee-b5d0cdef9913.png)

I launched dreamview as usual and lunched control, planning and routing modules and chosen a route and the car starts moving. 
Then I started debugging control module using 
`sudo gdb -q bazel-bin/modules/control/control -p <number id of process>`
in this case the debugging paused at first breakpoint alwasy in spite of the second breakpoint that i have added indicates to  controller agent (that means it ignored it) as shown in screenshot below
![Screenshot from 2021-01-21 14-38-08](https://user-images.githubusercontent.com/73900127/105346498-b7488e00-5bf6-11eb-8e91-cb762edef9f0.png)

In this situation I have stopped debugging process and turned off the control module in dreamview and again turn it on and started new debugging process for control module (i.e. I didn't stop the dreamview at all), In this case, the debugging process paused at both breakpoints and that means the control module executes controller_agent and by continuing the debugging process I have seen that the used controller is as expected (Lat+Lon).
![Screenshot from 2021-01-21 14-32-41](https://user-images.githubusercontent.com/73900127/105346998-55d4ef00-5bf7-11eb-84e6-a78bf6d2b8c1.png)

Could you please explain why the controller_agent wasn't executed in first case? and how I can fix this problem?

**Comment by MuhammadUbuntu** [comment]
I will add more information that i have gotten.
1- I have changed Lat+Lon to MPC and is has been worked as expected i.e. the controller agent used MPC.
2- I have launched control module after that the car started moving and the problem above was solved.

In summery, the problem is:
When I launched control module **before** the car starts moving, the `controller_agent` wasn't executed.
When I launched control module **after** the car starts moving, the `controller_agent `was executed.
